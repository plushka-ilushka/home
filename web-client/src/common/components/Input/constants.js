const INPUT_TYPES = {
  id: this.name,
  name: this.name,
  type: this.type,
  value: this.value,
  onChange: this.handleChange,
  placeholder: this.placeholder
};

export default INPUT_TYPES;
