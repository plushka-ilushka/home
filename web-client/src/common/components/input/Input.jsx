import React from 'react';
import PropTypes from 'prop-types';

import INPUT_TYPES from './constants';

import './input.scss';


export default function Input(props) {
  const {
    className,
    name,
    type,
    value,
    onChange,
    placeholder,
    ...restProps
  } = props;

  return (
    <input
      className={className}
      id={name}
      name={name}
      type={type}
      value={value}
      onChange={onChange}
      placeholder={placeholder}
      {...restProps}
    />
  );
}

Input.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
};

Input.defaultProps = {
  type: INPUT_TYPES.TEXT,
  className: undefined,
  name: undefined,
  placeholder: undefined,
  id: undefined,
  value: undefined,
  onChange: undefined,
};
