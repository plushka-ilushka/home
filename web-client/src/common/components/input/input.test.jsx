import React from 'react';
import { mount } from 'enzyme';
import Input from './Input';

describe('Input Component', () => {
  it('Input Snaphot', () => {
    expect(mount(<Input />)).toMatchSnapshot();
  });
});
