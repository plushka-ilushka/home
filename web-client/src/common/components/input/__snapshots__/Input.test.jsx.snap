// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Input Snaphot 1`] = `
<Input
  type="text"
>
  <input
    className="input"
    type="text"
  />
</Input>
`;
